scipy/special/tests/test_ellip_harm.py

```diff
@@ -271,3 +271,11 @@ def ellip_harm_known(h2, k2, n, p, s):
     points = np.array(points)
     assert_func_equal(ellip_harm, ellip_harm_known, points, rtol=1e-12)
 
+
+def test_ellip_harm_invalid_p():
+    # Regression test. This should return nan.
+    n = 4
+    # Make p > 2*n + 1.
+    p = 2*n + 2
+    result = ellip_harm(0.5, 2.0, n, p, 0.2)
+    assert_equal(result, np.nan)
```

Comment:
`assert np.isnan(result)` (or `np.assert_`) would be clearer here I think—avoids having to go look up whether `nan`s are going to be considered equal or not.

Reply:
Hmmm... I thought it was understood that numpy's `assert_equal` is specifically designed to understand `nan`, and I would be OK if someone new to the numpy testing library had to double-check the documentation to see how `nan` is handled.  But I just checked the docstring for [`assert_equal`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_equal.html), and it does not mention `nan`, so anyone new to this function would have to dig even further (grumble, grumble).  I'll change the test (and maybe submit a documentation change to numpy 😄 ).

Reply:
Fixed.